state/devices_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state_test
+
+import (
+	gc "gopkg.in/check.v1"
+
+	jc "github.com/juju/testing/checkers"
+
+	"github.com/juju/juju/core/devices"
+	"github.com/juju/juju/state"
+)
+
+type DevicesStateSuiteBase struct {
+	ConnSuite
+
+	series        string
+	st            *state.State
+	deviceBackend *state.DeviceBackend
+}
+
+func (s *DevicesStateSuiteBase) SetUpTest(c *gc.C) {
+	s.ConnSuite.SetUpTest(c)
+
+	var err error
+
+	if s.series == "kubernetes" {
+		s.st = s.Factory.MakeCAASModel(c, nil)
+		s.AddCleanup(func(_ *gc.C) { s.st.Close() })
+	} else {
+		s.st = s.State
+		s.series = "quantal"
+	}
+	s.deviceBackend, err = state.NewDeviceBackend(s.st)
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *DevicesStateSuiteBase) AddTestingCharm(c *gc.C, name string) *state.Charm {
+	return state.AddTestingCharmForSeries(c, s.st, s.series, name)
+}
+
+func makeDeviceCons(t devices.DeviceType, count int64) devices.Constraints {
+	return devices.Constraints{Type: t, Count: count}
+}
+
+type DevicesStateSuite struct {
```

Comment:
Since we use a base suite and handle caas and iaas (even though no iaas yet), should call this CAASDevicesSuite